atom/browser/lib/rpc-server.coffee

```diff
@@ -10,6 +10,7 @@ valueToMeta = (sender, value) ->
   meta.type = 'buffer' if Buffer.isBuffer value
   meta.type = 'value' if value is null
   meta.type = 'array' if Array.isArray value
+  meta.type = 'promise' if value? and value.constructor.name is 'Promise'
```

Comment:
> You are actually creating a new promise for that even if it's not a promise

Thanks @steelbrain ! I changed the implementation to use constructor name.

Looks like this works even for non-standard Promises like bluebird.